services/todo-legacy/controllers/todo-controller.api.ts

```diff
@@ -0,0 +1,225 @@
+export const def = {
+  basePath: "/",
+  paths: {
+    "/todo": {
+      get: {
+        "x-operation-name": "getTodo",
+        parameters: [
+          {
+            name: "title",
+            in: "query",
+            description:
+              "The title of the todo entry.",
+            required: false,
+            type: "string",
+            format: "JSON"
+          },
+          {
+            name: "filter",
+            in: "query",
+            description:
+              "The title of the todo entry.",
+            required: false,
+            type: "object",
+            format: "JSON"
+          },
+        ],
+        responses: {
+          200: {
+            schema: {
+              id: {
+                type: "number",
+                description: "The ID number of the Todo entry."
+              },
+              title: {
+                type: "string",
+                description: "The title of the Todo entry."
+              },
+              body: {
+                type: "string",
+                description: "The body of the Todo entry."
+              }
+            }
+          }
+        }
+      },
+      post: {
+        "x-operation-name": "createTodo",
+        parameters: [
+          {
+            name: "data",
+            in: "body",
+            description: "The Todo model instance.",
+            required: true,
+            format: "JSON"
+          }
+        ],
+        responses: {
+          201: {
+            schema: {
+              id: {
+                type: "number",
+                description: "The ID number of the Todo entry."
+              },
+              title: {
+                type: "string",
+                description: "The title of the Todo entry."
+              },
+              body: {
+                type: "string",
+                description: "The body of the Todo entry."
+              }
+            }
+          }
+        }
+      },
+      delete: {
+        "x-operation-name": "deleteTodo",
+        parameters: [
+          {
+            name: "title",
+            in: "query",
+            description:
+              "The criteria used to narrow down the number of customers returned.",
+            required: false,
+            type: "string",
+            format: "JSON"
+          }
+        ],
+        responses: {
+          204: {
+            description: "The resource has been deleted." 
+          }
+        }
+      }
+    },
+    "/todo/{id}": {
+      get: {
+        "x-operation-name": "getTodoById",
+        parameters: [
+          {
+            name: "id",
+            in: "path",
+            description:
+              "The criteria used to narrow down the number of customers returned.",
+            required: false,
+            type: "string",
+            format: "JSON"
+          }
+        ],
+        responses: {
+          200: {
+            schema: {
+              id: {
```

Comment:
The schema describing the Todo object is repeated in too many places. Have you considered moving the definition to `#/definitions/Todo` and then referencing it via `$ref` here?